exported: `main` package

symbols declared in `main` package [cannot be imported](https://go.dev/ref/spec#Program_execution).

It's the same for the content of `_test` files.

Here is a repository

https://github.com/CodeWithEmad/gh-fork-cleanup/blob/v1.1.0/main.go

It contains exported structures.

Right now, revive 1.10.0 reports the following errors

```
main.go:16:6: exported: exported type Owner should have comment or be unexported (revive)
type Owner struct {
     ^
main.go:21:6: exported: exported type Repo should have comment or be unexported (revive)
type Repo struct {
```

Test files are already excluded in `exported` rule
https://github.com/mgechev/revive/blob/v1.10.0/rule/exported.go#L114

There are multiple ways to consider this:
- `exported` should ignore a file when a file in `main`
- `exported` should report that these structures are exported in a package that cannot be exported.
- other ideas we may think about.

--- a/rule/exported.go
+++ b/rule/exported.go
@@ -110,11 +110,11 @@ func (r *ExportedRule) Configure(arguments lint.Arguments) error {
 
 // Apply applies the rule to given file.
 func (r *ExportedRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
-	var failures []lint.Failure
-	if file.IsTest() {
-		return failures
+	if !file.IsImportable() {
+		return nil
 	}
 
+	var failures []lint.Failure
 	walker := lintExported{
 		file: file,
 		onFailure: func(failure lint.Failure) {

--- a/test/exported_test.go
+++ b/test/exported_test.go
@@ -47,3 +47,7 @@ func TestCheckDirectiveComment(t *testing.T) {
 func TestCheckDeprecationComment(t *testing.T) {
 	testRule(t, "exported_issue_1231", &rule.ExportedRule{}, &lint.RuleConfig{})
 }
+
+func TestExportedMainPackage(t *testing.T) {
+	testRule(t, "exported_main", &rule.ExportedRule{}, &lint.RuleConfig{})
+}

--- a/testdata/exported_main.go
+++ b/testdata/exported_main.go
@@ -1,0 +1,4 @@
+// Symbols declared in main package cannot be imported.
+package main
+
+type Foo struct{}

